projects/extension/sql/idempotent/013-loading.sql

```diff
@@ -0,0 +1,87 @@
+-------------------------------------------------------------------------------
+-- loading_column
+create or replace function ai.loading_column
+( column_name pg_catalog.name)
+returns pg_catalog.jsonb
+as $func$
+    select json_object
+    ( 'implementation': 'column'
+    , 'config_type': 'loading'
+    , 'column_name': column_name
+    )
+$func$ language sql immutable security invoker
+set search_path to pg_catalog, pg_temp
+;
+
+-------------------------------------------------------------------------------
+-- loading_uri
+create or replace function ai.loading_uri
+( column_name pg_catalog.name)
+returns pg_catalog.jsonb
+as $func$
+    select json_object
+    ( 'implementation': 'uri'
+    , 'config_type': 'loading'
+    , 'column_name': column_name
+    )
+$func$ language sql immutable security invoker
+set search_path to pg_catalog, pg_temp
+;
+
+-------------------------------------------------------------------------------
+-- _validate_loading
+create or replace function ai._validate_loading
+( config pg_catalog.jsonb
+, source_schema pg_catalog.name
+, source_table pg_catalog.name
+) returns void
+as $func$
+declare
+    _config_type pg_catalog.text;
+    _implementation pg_catalog.text;
+    _column_name pg_catalog.name;
+    _found pg_catalog.bool;
+    _column_type pg_catalog.text;
+begin
+    if pg_catalog.jsonb_typeof(config) operator(pg_catalog.!=) 'object' then
+        raise exception 'loading config is not a jsonb object';
+end if;
+
+    _config_type = config operator(pg_catalog.->>) 'config_type';
+    if _config_type is null or _config_type operator(pg_catalog.!=) 'loading' then
+        raise exception 'invalid config_type for loading config';
+end if;
+
+    _implementation = config operator(pg_catalog.->>) 'implementation';
+    if _implementation is null or _implementation not in ('column', 'uri') then
+        raise exception 'invalid loading config implementation';
+end if;
+
+    _column_name = config operator(pg_catalog.->>) 'column_name';
+     if _column_name is null then
+        raise exception 'invalid loading config, missing column_name';
+end if;
+
+    select y.typname into _column_type
```

Comment:
@jgpruitt As we need to get the column type for later do proper validations based on it, I changed this query to just get the column type. Now the count(*) > 0 check is not needed. I had to remove the `strict` because the select could potentially return more than 1 row in this case now, however I didn't add a `limit 1` because I believe this won't ever happen (to return more than 1 result) right? If you think the `limit 1` is needed as consistency check, I can add it.

Is there anything else I'm not considering? 🙏

Reply:
Then it is fine to leave it as it is now with my change.

Reply:
done, thanks!